src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathSingleFilter.java

```diff
@@ -26,11 +26,377 @@
+ * {@code ancestor} axis is used to iterate all ancestor nodes of the current
+ * node with type {@code METHOD_DEF} and name <i>throwsMethod</i>.
+ * Please read more about xpath axes at
+ * <a href="https://www.w3schools.com/xml/xpath_axes.asp">W3Schools Xpath Axes</a>.
+ * </p>
+ * <pre>
+ * &lt;module name="SuppressionXpathSingleFilter"&gt;
+ *   &lt;property name="checks" value="IllegalThrows"/&gt;
+ *   &lt;property name="query" value="//LITERAL_THROWS/IDENT[
+ *           ..[@text='RuntimeException'] and ./ancestor::METHOD_DEF[@text='throwsMethod']]"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Code Example:
+ * </p>
+ * <pre>
+ * public class InputTest {
+ *   public void throwsMethod() throws RuntimeException { // violation will be suppressed
+ *   }
+ *
+ *   public void sampleMethod() throws RuntimeException { // will throw violation here
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * The following sample demonstrates how to suppress all violations for method
+ * itself and all descendants. {@code descendant-or-self} axis iterates through
+ * current node and all children nodes at any level. Keyword {@code node()}
+ * selects node elements. Please read more about xpath syntax at
+ * <a href="https://www.w3schools.com/xml/xpath_syntax.asp">W3Schools Xpath Syntax</a>.
+ * </p>
+ * <pre>
+ * &lt;module name="SuppressionXpathSingleFilter"&gt;
+ *   &lt;property name="checks" value=".*"/&gt;
+ *   &lt;property name="query" value="//METHOD_DEF[@text='TestMethod1']
+ *         /descendant-or-self::node()"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Code Example:
+ * </p>
+ * <pre>
+ * public class TestClass {
+ *   public void TestMethod1() { // OK
+ *     final int num = 10; // OK
+ *   }
+ *
+ *   public void TestMethod2() { // violation of the MethodName check,
+ *                               // name 'TestMethod2' must match pattern '^[a-z](_?[a-zA-Z0-9]+)*$'
+ *     final int num = 10; // violation of the LocalFinalVariableName check,
+ *                         // name 'num' must match pattern '^[A-Z][A-Z0-9]*$'
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * The following example is an example of what checks would be suppressed while
+ * building Spring projects with checkstyle plugin. Please find more information at:
+ * <a href="https://github.com/spring-io/spring-javaformat">spring-javaformat</a>
+ * </p>
+ * <pre>
+ * &lt;module name="SuppressionXpathSingleFilter"&gt;
+ *   &lt;property name="files" value="[\\/]src[\\/]test[\\/]java[\\/]"/&gt;
+ *   &lt;property name="checks" value="Javadoc*"/&gt;
+ * &lt;/module&gt;
+ * &lt;module name="SuppressionXpathSingleFilter"&gt;
+ *   &lt;property name="files" value=".*Tests\.java"&gt;
+ *   &lt;property name="checks" value="Javadoc*"&gt;
+ * &lt;/module&gt;
+ * &lt;module name="SuppressionXpathSingleFilter"&gt;
+ *   &lt;property name="files" value="generated-sources"&gt;
+ *   &lt;property name="checks" value="[a-zA-Z0-9]*"&gt;
+ * &lt;/module&gt;
+ *
+ * </pre> @since 8.18
```

Comment:
Move blank line before to be between `</pre>` and `@since`.

Reply:
It sounds like good Check idea

Reply:
@rnveach done
@romani done #6929